ferrocene/ci/split-tasks.py

```diff
@@ -122,6 +122,33 @@ def __lt__(self, other):
             ],
         ),
 
+        # like 'test:compiletest' minus the ONLY_HOSTS test groups
+        # the ONLY_HOSTS tests will run on the host even when
+        # `--target $NOT_THE_HOST` is passed to `x.py test`
+        # this avoids re-running tests on the same host triple across
+        # different CI jobs
+        "compiletest-no-only-hosts": find_all_compiletests(
```

Comment:
Can we double-check this doesn't accidentally exclude jobs from other runners?

Reply:
this change adds duplicated `--exclude` arguments to the output of `ferrocene/ci/split-tasks.py test` but the command below produces the same output before and after this PR

``` console
$ ferrocene/ci/split-tasks.py test | tr ' ' '\n' | sort -u
```

Reply:
This may be cause for concern still.

It may be more appropriate to create a new task category and add this subtask...

Eg:

![image](https://github.com/user-attachments/assets/baa57b47-95d6-4e87-944a-03863e4cb678)

```bash
$ python3 ferrocene/ci/split-tasks.py qnx:compiletest-no-only-hosts
tests/assembly tests/codegen tests/codegen-units tests/coverage tests/crashes tests/debuginfo tests/incremental tests/mir-opt tests/run-make tests/run-pass-valgrind tests/ui

$ python3 ferrocene/ci/split-tasks.py test
--exclude=ferrocene-check-document-signatures --exclude=library/alloc --exclude=library/core --exclude=library/std --exclude=library/test --exclude=src/bootstrap --exclude=src/tools/linkchecker --exclude=src/tools/tidy --exclude=tests/assembly --exclude=tests/codegen --exclude=tests/codegen-units --exclude=tests/coverage --exclude=tests/coverage-run-rustdoc --exclude=tests/crashes --exclude=tests/debuginfo --exclude=tests/incremental --exclude=tests/mir-opt --exclude=tests/pretty --exclude=tests/run-make --exclude=tests/run-pass-valgrind --exclude=tests/rustdoc --exclude=tests/rustdoc-js --exclude=tests/rustdoc-js-std --exclude=tests/rustdoc-json --exclude=tests/rustdoc-ui --exclude=tests/ui --exclude=tests/ui-fulldeps
```

I believe a new category would help avoid the double parameter behavior.

Reply:
moved compiletests-no-only-hosts into a separate category